src/python/gem5/components/boards/abstract_board.py

```diff
@@ -162,7 +178,13 @@ def get_cache_line_size(self) -> int:
 
         :returns: The size of the cache line size.
         """
-        return self.cache_line_size
+        if (line_size := getattr(self, "cache_line_size", None)) is not None:
+            return line_size
+        else:
+            # TODO: is it better to just return 0 (or some other default value) if the attribute is not set?
+            raise Exception(
```

Comment:
`cache_line_size` is a parameter of `System`, so maybe this function should be in the `AbstractSystemBoard`

Reply:
That makes sense, should I do that?
The reason I avoided that approach was to prevent api changes.

Reply:
yeah, I did some testing (process: move the function to AbstractSystemBoard, run mypy to see what breaks, fix some of the new errors, run mypy again, ...) and ... there's a lot of things that would have to be refactored (so much so I'd probably put it in a new pr). It's definitely possible, if it makes sense it's probably a good idea, and I'm willing to do that work, but it would likely be a breaking change.

Reply:
TL;DR, I think this change is worth doing, but I'd suggest we wait until after this PR is figured out and do it in a separate pr, should I open an issue?